lib/openhab/core/dsl/states.rb

```diff
@@ -0,0 +1,34 @@
+module OpenHAB
+  module Core
+    module DSL
+      module States
+        java_import org.openhab.core.model.script.actions.BusEvent
+
+        class StateStorage < Hash
+          def restore
+            BusEvent.restoreStates self
+          end
+
+          def restore_changes
+            BusEvent.restoreStates select { |item, value| item != value }
+          end
+
+          def changed?
+            each { |item, value| return true if item != value }
+            false
+          end
+        end
+
+        def store_states(*items, &block)
+          i = items.map {|item| item.is_a?(Group) ? item.group : item }
```

Comment:
rubocop is going to flag single letter variables. 
It would be ruby normal to just shadow `items` here.
```
items = items.map {|item| item.is_a?(Group) ? item.group : item }
```